addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,460 @@
+                        } else {
+                            logger.debug("Got name update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_STATUS_MODE:
+                        if (command instanceof StringType) {
+                            setMowerMode(command);
+                        } else {
+                            logger.debug("Got job remote start update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_MOWER_STATUS_STARTED:
+                        if (command instanceof OnOffType) {
+                            handleStartStop((OnOffType) command);
+                        } else {
+                            logger.debug("Got stopped update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_REMOTE_START:
+                        if (command instanceof StringType) {
+                            setRemoteStartJobSetting(command);
+                        } else {
+                            logger.debug("Got job remote start update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_AFTER_MODE:
+                        if (command instanceof StringType) {
+                            setAfterModeJobSetting(command);
+                        } else {
+                            logger.debug("Got job after mode update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_START:
+                        if (command instanceof StringType) {
+                            setStartJobSetting(command);
+                        } else {
+                            logger.debug("Got job start update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_END:
+                        if (command instanceof StringType) {
+                            setEndJobSetting(command);
+                        } else {
+                            logger.debug("Got job end update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                }
+
+            }
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
well, there are cases where the mower exits wifi range during an update leading to an exception, setting the thing offline. No, if the command is sent again, and it was successful, the mower would be online again. But of course it also is updated by the polling thread. Maybe could maybe removed here. But does it any harm? Because the polling thread might just poll every 30s. So to wait 30s in case the mower just exited range shortly is also bad, if the command is retired immediately and also successful..